From 019d1f04b83db8cd4d5a9381d5ef0c90e946cd05 Mon Sep 17 00:00:00 2001
From: Phong Man Tran <phong.tran.wh@rvc.renesas.com>
Date: Mon, 6 Jul 2015 14:48:50 +0700
Subject: [PATCH] rcar-gen3: add readme for building with Linaro Gcc

This patch add the dependent layer information, build instructions
for using Linaro Gcc.

Updated on Dec, 2016:
- Add H3ULCB
- Change supported branch from jethro to krogoth
- Change Maintainer information.

Signed-off-by: Phong Man Tran <phong.tran.wh@rvc.renesas.com>
Signed-off-by: Thao Nguyen <thao.nguyen.yb@rvc.renesas.com>
---
 meta-rcar-gen3/README.linaro | 122 +++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 122 insertions(+)
 create mode 100644 meta-rcar-gen3/README.linaro

diff --git a/meta-rcar-gen3/README.linaro b/meta-rcar-gen3/README.linaro
new file mode 100644
index 0000000..6542d43
--- /dev/null
+++ b/meta-rcar-gen3/README.linaro
@@ -0,0 +1,122 @@
+meta-rcar-gen3
+==============
+
+This layer provides that evaluation board is mounted ARM SoCs of Renesas
+Electronics, called the R-Car Generation 3. Currently, this supports
+board and the SoCs of the following:
+    - Board: Salvator-X   / SoC: R8A7795 (R-Car H3), R8A7796 (R-Car M3)
+    - Board: R-Car Starter Kit premier(H3ULCB)   / SoC: R8A7795
+
+Patches
+=======
+
+Please submit any patches for this layer to: takamitsu.honda.pv@renesas.com
+Please see the MAINTAINERS file for more details.
+
+Dependencies
+============
+
+This layer depends on:
+
+    URI: git://git.yoctoproject.org/poky
+    layers: meta, meta-yocto, meta-yocto-bsp
+    branch: krogoth
+
+    URI: git://git.yoctoproject.org/meta-renesas
+    layers: meta-rcar-gen3
+    branch: krogoth
+
+    URI: https://git.linaro.org/openembedded/meta-linaro.git
+    layers: meta-linaro-toolchain
+    branch: krogoth
+
+    URI: git://git.openembedded.org/meta-openembedded
+    layers: meta-oe
+    branch: krogoth
+
+Build Instructions
+==================
+
+The following instructions require a Poky installation (or equivalent).
+
+Initialize a build using the 'oe-init-build-env' script in Poky. e.g.:
+
+    $ source poky/oe-init-build-env
+
+After that, initialized configure bblayers.conf by adding meta-rcar-gen3 layer. e.g.:
+
+    BBLAYERS ?= " \
+        <path to layer>/poky/meta \
+        <path to layer>/poky/meta-yocto \
+        <path to layer>/poky/meta-yocto-bsp \
+        <path to layer>/meta-renesas/meta-rcar-gen3 \
+        <path to layer>/meta-linaro/meta-linaro-toolchain \
+        <path to layer>/meta-openembedded/meta-oe \
+    "
+
+To build a specific target BSP, configure the associated machine in local.conf:
+
+    MACHINE ??= "<supported board name>"
+
+Select the SOC
+
+    For H3: r8a7795
+    SOC_FAMILY = "r8a7795"
+
+    For M3: r8a7796
+    SOC_FAMILY = "r8a7796"
+
+Must change the package class to ipk files in local.conf:
+
+    PACKAGE_CLASSES ?= "package_ipk"
+
+Must Change the Gcc version in local.conf.
+
+    GCCVERSION="linaro-5.2"
+
+Configure for systemd init in local.conf:
+
+    DISTRO_FEATURES_append = " systemd"
+    VIRTUAL-RUNTIME_init_manager = "systemd"
+
+Build the target file system image using bitbake:
+
+    $ bitbake core-image-minimal
+
+After completing the images for the target machine will be available in the output
+directory 'tmp/deploy/images/<supported board name>'.
+
+Images generated:
+    * Image (generic Linux Kernel binary image file)
+    * Image-<machine name>.dtb (DTB for target machine)
+    * core-image-minimal-<machine name>.tar.bz2 (rootfs tar+bzip2)
+    * core-image-minimal-<machine name>.ext4  (rootfs ext4 format)
+    * core-image-minimal-<machine name>.cpio.gz (rootfs cpio+gz format)
+
+Build Instructions for SDK
+==========================
+This may be changed in the near feature. These instructions are tentative.
+
+Should define the staticdev in SDK image feature for installing the static libs
+to SDK in local.conf.
+
+    SDKIMAGE_FEATURES_append = " staticdev-pkgs"
+
+Use bitbake -c populate_sdk for generating the toolchain SDK:
+For 64-bit target SDK (aarch64):
+
+    $ bitbake core-image-minimal -c populate_sdk
+
+The SDK can be found in the output directory 'tmp/deploy/sdk'
+    * poky-glibc-x86_64-core-image-minimal-aarch64-toolchain-x.x.sh
+
+Usage of toolchain SDK:
+Install the SDK to the default: /opt/poky/x.x
+(x.x is YP version, i.e. 2.1.)
+For 64-bit target SDK:
+
+    $ sh poky-glibc-x86_64-core-image-minimal-aarch64-toolchain-x.x.sh
+
+For 64-bit application use environment script in /opt/poky/x.x
+
+    $ source /opt/poky/x.x/environment-setup-aarch64-poky-linux
-- 
1.9.1

